family tree

KSerzh opened the pull request:
HomeWorke_familyTree_OOP

KSerzh committed:
family tree

Liberate520 commented on src/Main.java:
```diff
@@ -0,0 +1,90 @@
+
+class Person {
+    private String name;
+    private int age;
```
использовать дату рождения и смерти вместо возраста, а возраст высчитывать в методе

Liberate520 commented on src/Main.java:
```diff
@@ -0,0 +1,90 @@
+class Person {
+    private String name;
+    private int age;
+    private List<Person> parents;
```
С детьми вижу методы, а как родителей добавлять?

Liberate520 commented on src/Main.java:
```diff
@@ -0,0 +1,90 @@
+        }
+        return null;
+    }
+
```
добавить toString

KSerzh committed:
Split Classes.
Calculate age from date of birth and date of death.
Added methods for parents and toString

KSerzh committed:
Createв writer/save FamilyTree in fale

KSerzh committed:
Createв writer/save FamilyTree in fale

KSerzh committed:
Merge remote-tracking branch 'origin/master'

KSerzh committed:
Merge remote-tracking branch 'origin/master'

KSerzh committed:
Merge remote-tracking branch 'origin/master'

KSerzh committed:
FamilyTree sorted by name. FamilyTree sorted by birth date

KSerzh committed:
FamilyTree sorted by name. FamilyTree sorted by birth date

KSerzh committed:
Merge remote-tracking branch 'origin/master'

KSerzh committed:
Parameterized class. Switch (choice) 1, 2, 3, 4, 5, 6, 7, 8. Add interface: HasName, HasBirthDate, HasRelations. we can create a Dog class outside the project that implements the same interfaces (HasName, HasBirthDate, HasRelations) and use it with FamilyTree<Dog>

Liberate520 commented on src/familytree/FamilyTree.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    public List<T> getChildren(T person) {
+        if (person instanceof HasRelations) {
```
не должны использоваться операторы instanceof. Сама идея дженериков как раз для того, чтоб не надо было проверять типы и делать приведение типов

KSerzh committed:
MVP

Liberate520 commented on src/presenter/Presenter.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.LocalDate;
+
+public class Presenter {
+    private FamilyTree<Person> familyTree;
```
Нужно создать класс Сервис для объединения работы модели. Сервис будет управлять деревом, билдером и объектом для реализации сохранений

Liberate520 commented on src/presenter/Presenter.java:
```diff
@@ -0,0 +1,45 @@
+
+    public void addPerson(String name, LocalDate birthDate, LocalDate deathDate, String genderStr) {
+        Gender gender = Gender.valueOf(genderStr.toUpperCase());
+        Person person = new Person(name, birthDate, deathDate, gender);
```
презентер не должен создавать объекты модели

KSerzh committed:
SOLID

Liberate520 commented on src/service/TreeSorter.java:
```diff
@@ -0,0 +1,16 @@
+
+public class TreeSorter<T extends Person> {
+
+    public void sortByName(List<T> members) {
+        members.sort(Comparator.comparing(Person::getName));
+    }
+
+    public void sortByBirthDate(List<T> members) {
+        members.sort(Comparator.comparing(Person::getBirthDate));
+    }
```
методы перенести в класс дерева. Текущий класс удалить

Liberate520 commented on src/view/ConsoleTreeView.java:
```diff
@@ -0,0 +1,88 @@
+    public void start() {
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            System.out.println("Enter command (add, find, display, sortName, sortBirth, save, load, exit): ");
```
нарушает второй принцип солид

Liberate520 commented on src/view/ConsoleTreeView.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    private void handleCommand(String command, Scanner scanner) {
+        switch (command.toLowerCase()) {
+            case "add":
```
свич кейс нарушает второй принцип солид

Liberate520 replied:
нет метода для создания родственной связи

KSerzh committed:
comment in git 2024.08.16

KSerzh committed:
addPerson(name, birthDate, deathDate, genderStr, spouseName, childrenNames, fatherName, motherName)

KSerzh committed:
tree_my

KSerzh committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	tree_my

Liberate520 commented on src/writer/TreeStorage.java:
```diff
@@ -0,0 +1,10 @@
+import familytree.FamilyTree;
+import person.Person;
+
+public interface TreeStorage<T extends Person> {
```
Зависимость параметра от Person это ошибка
